pkg/ingester/index/multi.go

```diff
@@ -0,0 +1,123 @@
+package index
+
+import (
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/querier/astmapper"
+	"github.com/grafana/loki/pkg/storage/config"
+)
+
+type periodIndex struct {
+	time.Time
+	idx int // address of the index to use
```

Comment:
a bit confused about what this represents. it seems to be a type more than an address to me, ie 0 is a regular inverted index, whereas 1 is a tsdb inverted index. could we reflect that in the name and/or comment?

Reply:
or is this the index in the `indices` property on `Multi`? If so, and there can only be 2, I'm curious if instead of storing these in a fixed length slice with magic indices, maybe `Multi` should have a field for each type?

Reply:
That's a fair point. I wrote it like this in case we had period configs like `[boltdb, tsdb, boltdb, tsdb]`, in which case we'd want the first period to point to the index of the `boltdb` inverted index, the second to point to the `tsdb` inverted index and so on. Decoupling the period from the index it points to allows us to instantiate a maximum of one inverted index for each type, regardless of the number of period configs.

The way it's written now will allow us to add other index types in the future if necessary. Does that make sense?

Reply:
it does, but we could also come back and update this struct with the necessary types when we get to that point. just feels like we're overloading slice indexing as a mechanism for expressing type, whereas there are clearer ways to do that.

that being said I have no interest in dying on this hill, and I'm cool with keeping it this way if you think it's more flexible.